modules/product/src/Form/AddToCartForm.php

```diff
@@ -166,4 +173,224 @@ function submitForm(array &$form, FormStateInterface $form_state) {
     ]));
   }
 
+  /**
+   * Builds the variation selection form.
+   *
+   * @param array $form
+   *   An associative array containing the structure of the form.
+   * @param \Drupal\Core\Form\FormStateInterface $form_state
+   *   The current state of the form.
+   * @param \Drupal\commerce_product\Entity\ProductInterface $product
+   *   The product that has the variations.
+   */
+  protected function variationForm(array &$form, FormStateInterface $form_state, ProductInterface $product) {
+    $default_variation = $form_state->get('default_variation');
+    $attribute_names = [];
+    $unchanged_attributes = [];
+    $user_input = $form_state->getUserInput();
+
+    $variations = $product->variations->referencedEntities();
```

Comment:
This doesn't exist? Only in ProductVariationManager? So use the service instead of the field directly?